Add Mistral Large 3 and Ministral 3

juliendenize opened the pull request:
## Purpose

This PR adds support to Mistral-Large-3 and Ministral-3.

## Test Plan

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

juliendenize committed:
Add Mistral Large 3

Signed-off-by: Julien Denize <julien.denize@mistral.ai>

juliendenize committed:
Merge branch 'main' into add_mistral_large_3

mgoin reviewed (commented):
🥳 Make sure to update the supported model page https://github.com/vllm-project/vllm/blob/main/docs/models/supported_models.md and the testing registry https://github.com/vllm-project/vllm/blob/main/tests/models/registry.py with is_available_online=False for now

juliendenize committed:
Merge branch 'main' into add_mistral_large_3

juliendenize commented on vllm/model_executor/layers/mla.py:
```diff
@@ -159,6 +160,9 @@ def forward_native(
             )
 
+        if llama_4_scaling is not None:
+            q *= llama_4_scaling
```
Just a choice but no strong opinion about this that i can think of now. We also did in llama.py would it be necessary to refactor now or could it be in later pr ?

juliendenize committed:
Wip

Signed-off-by: Julien Denize <julien.denize@mistral.ai>

juliendenize committed:
change supported model order

Signed-off-by: Julien Denize <julien.denize@mistral.ai>

juliendenize committed:
spelling

Signed-off-by: Julien Denize <julien.denize@mistral.ai>

juliendenize commented on vllm/entrypoints/openai/tool_parsers/mistral_tool_parser.py:
```diff
@@ -80,7 +80,7 @@ def __init__(self, tokenizer: TokenizerLike):
         if _is_fn_name_regex_support(self.model_tokenizer):
             self.fn_name_regex = re.compile(
-                r"([a-zA-Z0-9_-]+)(\{[\s\S]*?\})(?=\s*$|,|\s)", re.DOTALL
+                r"([a-zA-Z0-9_-]+)(\{[\s\S]*?\})(?=\s*$|,|\s)?", re.DOTALL
```
Had to add this to make sure tool calls worked as expected for some examples, will rerun some xp to see if i didn't break anything there as we didn't author this.

juliendenize replied:
Looks good on the examples I launched so this looks benign

juliendenize committed:
Merge branch 'main' into add_mistral_large_3

juliendenize committed:
Fix

Signed-off-by: Julien Denize <40604584+juliendenize@users.noreply.github.com>

juliendenize committed:
Merge branch 'main' into add_mistral_large_3

juliendenize committed:
Add broader tooling support

Signed-off-by: Julien Denize <julien.denize@mistral.ai>

juliendenize committed:
Add batch_decode

Signed-off-by: Julien Denize <julien.denize@mistral.ai>

juliendenize committed:
wip

Signed-off-by: Julien Denize <julien.denize@mistral.ai>

patrickvonplaten commented on vllm/transformers_utils/configs/eagle.py:
```diff
@@ -41,7 +41,9 @@ def __init__(
         # LlamaForCausalLM -> Eagle3LlamaForCausalLM
         # LlamaForCausalLMEagle3 -> LlamaForCausalLMEagle3
-        if method == "eagle":
+        if method is None:
```
Should we just revert this to make sure the PR gets in?

zou3519 replied:
Might be too late to revert all of https://github.com/vllm-project/vllm/pull/26468, but I can put the use_diff=False back

juliendenize committed:
wip

Signed-off-by: Julien Denize <julien.denize@mistral.ai>

ywang96 committed:
Merge branch 'main' into add_mistral_large_3

mickaelseznec committed:
fix: eagle and no --enforce-eager

Hashing the config can crash because constructor is called with default
arguments only.

Pass use_diff=false to avoid this behavior

Signed-off-by: Mickael Seznec <mickael@mistral.ai>

mickaelseznec committed:
fix: redo default arg in eagle

Signed-off-by: Mickael Seznec <mickael@mistral.ai>

ywang96 reviewed (approved):
Amazing! Thanks for all the work and look forward to the release!

Approving as the rest of the comments can be addressed in follow-up PRs.

mickaelseznec committed:
rename: triton moe config file

Signed-off-by: Mickael Seznec <mickael@mistral.ai>

mickaelseznec committed:
fix: loading mistral eagle

Signed-off-by: Mickael Seznec <mickael@mistral.ai>

mickaelseznec committed:
fix: eagle & compile

only fix for eagle config, avoid larger impact on the codebase

Signed-off-by: Mickael Seznec <mickael@mistral.ai>

ywang96 committed:
add and skip

Signed-off-by: Roger Wang <hey@rogerw.io>

mickaelseznec committed:
fix: default args for llam4_scaling

Signed-off-by: Mickael Seznec <mickael@mistral.ai>

mickaelseznec committed:
fix: add eagle model in test registry

Signed-off-by: Mickael Seznec <mickael@mistral.ai>

ywang96 committed:
Merge branch 'main' into add_mistral_large_3

ywang96 committed:
Merge branch 'main' into add_mistral_large_3
